packages/marshal/src/passStyleOf.js

```diff
@@ -141,10 +145,15 @@ const makePassStyleOf = passStyleHelpers => {
           if (inner === null) {
             return 'null';
           }
-          (isFrozen(inner)) ||
+          if (!isFrozen(inner)) {
             assert.fail(
-              X`Cannot pass non-frozen objects like ${inner}. Use harden()`,
+              // TypedArrays get special treatment in harden()
+              // and a corresponding special error message here.
+              isTypedArray(inner)
+                ? X`Cannot pass mutable typed arrays like ${inner}.`
```

Comment:
```suggestion
                ? X`Cannot pass typed arrays like ${inner}`
```

This will be a good place to recommend CopyBytes when they’re an option.